src/mongocxx/test/versioned_api.cpp

```diff
@@ -73,10 +73,29 @@ TEST_CASE("Verioned API, non-strict, for commands/features outside versioned API
     // Store it in the set of client options
     const auto client_opts =
         options::client{}
-            .server_api_opts(server_api_opts); // Set the version
+            .server_api_opts(server_api_opts); // Set the version and options
     // Create a new client with the options
     mongocxx::client client{client_uri, client_opts};
-    // End Versioned API Example 1
+    // End Versioned API Example 3
+}
+
+TEST_CASE("Versioned API, non-strict with deprecation errors") {
+    instance::current();
+    // Start Versioned API Example 4
+    using namespace mongocxx;
+    uri client_uri{"mongodb://localhost"};
+    // Create an option set for API v1
+    const auto server_api_opts =
+        options::server_api{options::server_api::version_from_string("1")}
+            .strict(false)              // Explicitly disable strict mode for the server API
```

Comment:
The pseudo code in CXX-2257 for `Versioned API Example 4` does not explicitly set the `strict` option. I think this line can be removed.